### PR TITLE
Added gcc-7

### PR DESCRIPTION
**Pull request by denzor200**

### CODE REVIEW

File: include/boost/pfr/detail/integer_sequence.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#define BOOST_PFR_DETAIL_INTEGER_SEQUENCE_HPP
+#pragma once
+
+#include <boost/pfr/detail/config.hpp>
+
+#if BOOST_PFR_USE_STD_INTEGRAL_SEQUENCE == 1
+#include <utility>
+#endif
+
+namespace boost { namespace pfr { namespace detail {
```
Comment (apolukhin) [inline include/boost/pfr/detail/integer_sequence.hpp:16]: Always include <utility> and check here for the __cpp_lib_integer_sequence feature macro. In that way there's less macro and creepy "do not include before " warnings

File: include/boost/pfr/detail/make_integer_sequence.hpp [hunk 2]
```diff
@@ -26,11 +26,8 @@ namespace boost { namespace pfr { namespace detail {
 
 #ifdef BOOST_PFR_USE_MAKE_INTEGER_SEQ_BUILTIN
 
-using std::integer_sequence;
-
-// Clang unable to use namespace qualified std::integer_sequence in __make_integer_seq.
```
Comment (apolukhin) [inline include/boost/pfr/detail/make_integer_sequence.hpp:31]: This comment has reasons :)
Put `using std::integer_sequence;` up above and do not use qualified name in `__make_integer_seq`

File: include/boost/pfr/detail/integer_sequence.hpp (hunk 1 above)
Comment (denzor200) [reply include/boost/pfr/detail/integer_sequence.hpp]: Not actual

File: include/boost/pfr/detail/make_integer_sequence.hpp (hunk 2 above)
Comment (denzor200) [reply include/boost/pfr/detail/make_integer_sequence.hpp]: Not actual

**Comment by denzor200** [comment]
Needs more informative error message in core17_generated.hpp

**Comment by denzor200** [comment]
@apolukhin i'm finished

File: test/Jamfile.v2 [hunk 3]
```diff
@@ -5,14 +5,21 @@
...
 #
 
+import os ;
 import python ;
 import testing ;
 import ../../config/checks/config : requires ;
 
+local USE_FALLBACK_FOR_BROKEN_STRUCTURED_BINDINGS = <define>BOOST_PFR_USE_CPP17=0 ;
+
+if ! [ os.environ BOOST_PFR_TEST_WITH_FALLBACK_FOR_BROKEN_STRUCTURED_BINDINGS ]
```
Comment (apolukhin) [inline test/Jamfile.v2:15]: This does not scales well to other compilers.

See `compiler_supports_loophole` and `loophole_detection.cpp`. Implement the structured binding check in a `structured_binding_detection.cpp` file and use that check.

**Review by apolukhin — CHANGES_REQUESTED** [review]

**Comment by denzor200** [comment]
```
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.5.0/../../../../include/c++/5.5.0/tuple:807:33: error: implicit instantiation of undefined template 'std::tuple_size<my_struct>'
    : integral_constant<size_t, tuple_size<_Tp>::value> { };
                                ^
./boost/pfr/detail/core17_generated.hpp:57:9: note: in instantiation of template class 'std::tuple_size<const my_struct>' requested here
  auto& [a,b,c] = val; // ====================> Boost.PFR: User-provided type is not a SimpleAggregate.
```
another day, troubles are the same

File: test/Jamfile.v2 (hunk 3 above)
Comment (denzor200) [reply test/Jamfile.v2]: not actual

**Comment by denzor200** [comment]
> ```
> /usr/bin/../lib/gcc/x86_64-linux-gnu/5.5.0/../../../../include/c++/5.5.0/tuple:807:33: error: implicit instantiation of undefined template 'std::tuple_size<my_struct>'
>     : integral_constant<size_t, tuple_size<_Tp>::value> { };
>                                 ^
> ./boost/pfr/detail/core17_generated.hpp:57:9: note: in instantiation of template class 'std::tuple_size<const my_struct>' requested here
>   auto& [a,b,c] = val; // ====================> Boost.PFR: User-provided type is not a SimpleAggregate.
> ```
> 
> another day, troubles are the same

i suppose, i know how to fix it https://godbolt.org/z/hWzsojGe6

**Comment by denzor200** [comment]
Closed because it's not actual.
Actual PR here: https://github.com/boostorg/pfr/pull/116

**Pull request closed without merging**